pkg/stacks/unpack.go

```diff
@@ -218,13 +238,97 @@ func (sp *StackPackage) IsNamespaced() bool {
 }
 
-// SetBehavior sets the Stack's Behavior configuration
-func (sp *StackPackage) SetBehavior(behavior unstructured.Unstructured) {
-	sp.Behavior = behavior
+// SetStackConfig sets the Stack's Behavior using StackConfig
+func (sp *StackPackage) SetStackConfig(config unstructured.Unstructured) {
+	sp.StackConfig = config
+}
+
+func (sp *StackPackage) createBehaviorController(sd v1alpha1.StackDefinitionBehavior) appsv1.DeploymentSpec {
+	// TODO(displague) where does the name come from? behavior? cmd line arg?
+	// sp.Stack.GetName() is not available when this runs
+	controllerDeploymentName := "tc-" + "something"
+	controllerContainerName := controllerDeploymentName
+	controllerImageName := "bymafmaf/templating-controller:0.0.1"
+
+	// TODO(displague) Where do these files reside within the Stack?
+	// are sp.baseDir or sd.Source.Path useful here?
+	// are we always mounting a remote source image?
+	behaviorDirName := "/"
+	behaviorContentsVolumeName := "behaviors"
+	behaviorMountPoint := "/behaviors"
+	spec := appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": controllerDeploymentName}},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{"app": controllerDeploymentName},
+			},
+			Spec: corev1.PodSpec{
+				RestartPolicy: corev1.RestartPolicyNever,
+				InitContainers: []corev1.Container{
+					{
+						Name:    "stack-package",
+						Image:   sd.Source.Image,
+						Command: []string{"cp", "-xR", behaviorDirName, behaviorMountPoint},
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      behaviorContentsVolumeName,
+								MountPath: behaviorMountPoint,
+							},
+						},
+					},
+				},
+				Containers: []corev1.Container{
+					{
+						Name:  controllerContainerName,
+						Image: controllerImageName,
+						Args: []string{
+							"manager",
+							"--resources-dir", behaviorMountPoint,
+							"--stack-definition-namespace", sp.Stack.GetNamespace(),
+							"--stack-definition-name", sp.Stack.GetName(),
+						},
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      behaviorContentsVolumeName,
+								MountPath: behaviorMountPoint,
+							},
+						},
+					},
+				},
+				Volumes: []corev1.Volume{
+					{
+						Name: behaviorContentsVolumeName,
+						VolumeSource: corev1.VolumeSource{
+							EmptyDir: &corev1.EmptyDirVolumeSource{},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return spec
+}
+
+// SetBehavior sets the Stack's Behavior
+// This is primarily for defining Template Stack behaviors
+func (sp *StackPackage) SetBehavior(sd v1alpha1.StackDefinitionBehavior) {
+	sd.DeepCopyInto(&sp.StackDefinition.Spec.Behavior)
+
+	// Set the Stack deployment, which will be copied to StackDefinition in
+	// Yaml()
+	// TODO(displague) what controller are we running?
+	spec := sp.createBehaviorController(sd)
+	sp.Stack.Spec.Controller.Deployment = &v1alpha1.ControllerDeployment{
+		Spec: spec,
```

Comment:
I'm getting `failed to create deployment: Deployment.apps "" is invalid: [metadata.name: Required value: name or generateName is required` I think you need to fill `Name`, too. At least that's what `(sp *StackPackage) SetInstall(install unstructured.Unstructured)` function does.